### PR TITLE
added grep command info

### LINKED ISSUE #13: Grep
I'll be working on Grep!

### PR DESCRIPTION
**Pull request by ahyerman**
Closes #13

### CODE REVIEW

File: _commands/basics/grep.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+---
+
+grep
+-------
+
+`grep` is used to search files for the occurence os string of characters matching a special pattern (sometimes called a regex or regular expression).
```
Comment (mterwill) [inline _commands/basics/grep.md]: Typo? Also note that grep doesn't support full regex – that's egrep. Would make note somewhere in here.

**Comment by mterwill** [comment]
I would also like to see an example of grep used to filter another (simple) command, as that could definitely benefit the target audience of this reference. Even something as simple as  ``ls -l | grep cpp``

**Comment by ahyerman** [comment]
fixed spelling typo. Added ls -l example. Using grep as a filter for commands to search for files. Explained process could be used to find processes as well. Linked grep to ls page.

**Comment by themoh** [comment]
A small section on the -e flag/egrep would be useful

**Comment by ahyerman** [comment]
Added recommended changes with an example of using egrep. Also removed regex from the top example to instead say string.

**Comment by themoh** [comment]
Looks really good so far. Add in a section on the -r flag also, and I think you should be good. @mterwill any other suggestions?

**Comment by ahyerman** [comment]
Added a section with the -r flag along with an example.

**Comment by ppannuto** [comment]
Looks good, thanks!

**Pull request merged**